src/main/java/macrobase/analysis/MacrobaseAnalyzer.java

```diff
@@ -0,0 +1,55 @@
+package macrobase.analysis;
+
+import macrobase.analysis.outlier.OutlierClassifier;
+import macrobase.analysis.result.MacrobaseAnalysisResult;
+import macrobase.analysis.summary.Summarizer;
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+import macrobase.ingest.DataIngester;
+import macrobase.transform.FeatureTransform;
+import org.apache.commons.collections4.iterators.IteratorIterable;
+
+import java.util.Iterator;
+
+public class MacrobaseAnalyzer implements Iterator<MacrobaseAnalysisResult> {
+
+    private DataIngester ingester;
+    private FeatureTransform source;
+    private OutlierClassifier outlierClassifier;
+    private Summarizer summarizer;
+    private int windowSize;
+
+    /**
+     * MacrobaseAnalyzer will create the following attributes in the constructor:
+     *  - DataIngester ingester, that gets the data, usually just initialize with a conf, based on conf
+     *  - FeatureTransform based on conf and ingester
+     * @param conf
+     */
+    public MacrobaseAnalyzer(MacroBaseConf conf) {};
+    // Example pseudo-code,
+    // ingester = constructIngester(conf);
+    // source = new FeatureTransform(conf, ingester);
+    // windowSize = conf.getWindowSize();
+    // outlierClassifier = constructOutlierClassifier(conf);
+    // Summarizer = new Summarizer(conf, outlierClassifier);
+
+    @Override
+    public MacrobaseAnalysisResult next() {
+
+        int index = 0;
+        for (Datum nextDatum : new IteratorIterable<>(source)) {
+            summarizer.observe(nextDatum);
```

Comment:
Oh I see, so the windowing would just be handled inside `Summarizer.next` rather than here? That makes sense. Although perhaps a little inconsistent since all the other iterators return/operate over a single datum.